### DISCUSSION TITLE
`vite build` is generating a broken production build

### DISCUSSION THREAD

**Discussion by iMrDJAi** [vikejs/vike#615] [Discussion]
This is caused by imports inside `index.page.server.ts` pointing to the wrong place.
In my case, some files from the backend (basically some MongoDB schemas) are getting built and included inside `./dist`. However, the source files are the ones used by the app entry point `server.ts`. So, there are 2 separate versions of them used in the production build, and since the schemas from `./dist` are not connected to the database, I'm getting the following error: `MongooseError: Cannot call setClient() with a MongoClient that you have not called connect() on yet.`.

What I've tried so far:

Using a dynamic import, instead of `import User from '../database/schemas/User'` I did `const User = (await import(process.cwd() + '/database/schemas/User')).default`. That worked in prod but it eventually broke the dev build. I tried messing up with `vite.config.ts`' aliases but nothing worked!

Any ideas? cc @brillout

**Comment by iMrDJAi** [comment]
Related: https://github.com/brillout/vite-plugin-ssr/issues/562
That wouldn't happen if the server code gets transpiled by Vite.
In the meanwhile, is there a workaround for this?

**Reply by brillout** [reply]
Using [vavite](https://github.com/cyco130/vavite) will make Vite transpile your server code.

**Reply by iMrDJAi** [reply]
Yes! vavite seems the answer! Could you please post this again so I can mark it as an answer?

**Reply by brillout** [reply]
I just marked parent as the answer.
